src/software/ai/navigator/navigator.cpp

```diff
@@ -2,303 +2,312 @@
-                                 ->RobotObstacleInflationFactor()
-                                 ->value() *
-                             ROBOT_MAX_RADIUS_METERS);
-            return Obstacle(rectangle);
-        default:
-            LOG(WARNING) << "Could not convert AvoidArea " << (int)avoid_area
-                         << " to obstacle";
-    }
+    Rectangle navigable_area(
+        Point(this->world.field().totalXLength(), this->world.field().totalYLength()),
+        this->world.field().totalXLength(), this->world.field().totalYLength());
 
-    return std::nullopt;
-}
+    auto path_objective_to_move_intent = generatePathObjectiveToMoveIntentMap();
 
-std::vector<std::unique_ptr<Primitive>> Navigator::getAssignedPrimitives(
-    const World &world, const std::vector<std::unique_ptr<Intent>> &assignedIntents)
-{
-    this->world              = world;
-    this->current_robot      = std::nullopt;
-    this->velocity_obstacles = {};
-    planned_paths.clear();
+    // get keyset from path_objective_to_move_intent and then plan paths
+    std::set<PathObjective> path_objectives;
+    for (const auto &p : path_objective_to_move_intent)
+    {
+        path_objectives.insert(p.first);
+    }
+    auto path_objective_to_path =
+        path_manager->getManagedPaths(path_objectives, navigable_area);
 
-    auto assigned_primitives = std::vector<std::unique_ptr<Primitive>>();
-    for (const auto &intent : assignedIntents)
+    for (const auto &path_objective_and_path : path_objective_to_path)
     {
-        intent->accept(*this);
-        if (this->current_robot)
+        auto path_objective = path_objective_and_path.first;
+        auto path           = path_objective_and_path.second;
+
+        // look for move intent
+        auto path_objective_and_intent_it =
+            path_objective_to_move_intent.find(path_objective);
+        if (path_objective_and_intent_it == path_objective_to_move_intent.end())
         {
-            if (this->current_robot->velocity().len() > 0.3)
-            {
-                this->velocity_obstacles.emplace_back(
-                    Obstacle::createVelocityObstacleWithScalingParams(
-                        this->current_robot->position(), this->current_destination,
-                        this->current_robot->velocity().len(),
-                        Util::DynamicParameters->getNavigatorConfig()
-                            ->RobotObstacleInflationFactor()
-                            ->value(),
-                        Util::DynamicParameters->getNavigatorConfig()
-                            ->VelocityObstacleInflationFactor()
-                            ->value()));
-            }
+            std::stringstream ss;
+            ss << "Failed to find intent associated with path objective that has robot id = "
+               << path_objective.robot_id;
+            LOG(WARNING) << ss.str();
 
-            this->current_robot = std::nullopt;
+            // assume no path
+            continue;
```

Comment:
What does the end behavior of this look like? Do we just give no primitive to that robot? A stop primitive?

Reply:
no primitive atm

Reply:
fixed with new arch we discussed